controllers/awsmachine_controller.go

```diff
@@ -701,7 +701,16 @@ func (r *AWSMachineReconciler) resolveUserData(machineScope *scope.MachineScope,
 	}
 
 	if machineScope.UseIgnition(userDataFormat) {
-		userData, err = r.ignitionUserData(machineScope, objectStoreSvc, userData)
+		ignitionStorageType := machineScope.AWSMachine.Spec.Ignition.StorageType
```

Comment:
Seems this could still produce a nil pointer exception? The above is an OR between the secret type and spec.ignition